src/EpisodeIntegrationService/ReceiveData/ReceiveData.cs

```diff
@@ -1,56 +1,212 @@
 using Microsoft.Extensions.Logging;
+using NHS.ServiceInsights.Common;
+using NHS.ServiceInsights.Model;
 using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
 using System.Text.Json;
-using NHS.ServiceInsights.Common;
+using System.Net;
+using CsvHelper;
+using System.Globalization;
 
-namespace NHS.ServiceInsights.EpisodeManagementService;
+namespace NHS.ServiceInsights.EpisodeIntegrationService;
 
 public class ReceiveData
 {
-
     private readonly ILogger<ReceiveData> _logger;
     private readonly IHttpRequestService _httpRequestService;
 
+    private readonly string[] episodesExpectedHeaders = new[] {"nhs_number","episode_id","episode_type","change_db_date_time","episode_date","appointment_made","date_of_foa","date_of_as","early_recall_date","call_recall_status_authorised_by","end_code","end_code_last_updated","bso_organisation_code","bso_batch_id","reason_closed_code","end_point","final_action_code"};
+
+    private readonly string[] subjectsExpectedHeaders = new[] {"change_db_date_time","nhs_number","superseded_nhs_number","gp_practice_code","bso_organisation_code","next_test_due_date","subject_status_code","early_recall_date","latest_invitation_date","removal_reason","removal_date","reason_for_ceasing_code","is_higher_risk","higher_risk_next_test_due_date","hr_recall_due_date","higher_risk_referral_reason_code","date_irradiated","is_higher_risk_active","gene_code","ntdd_calculation_method","preferred_language"};
+
     public ReceiveData(ILogger<ReceiveData> logger, IHttpRequestService httpRequestService)
     {
         _logger = logger;
         _httpRequestService = httpRequestService;
     }
+
     [Function("ReceiveData")]
-    public async Task Run(
-        [BlobTrigger("sample-container/{name}", Connection = "AzureWebJobsStorage")] Stream myBlob,
-        string name)
+    public async Task Run([BlobTrigger("sample-container/{name}", Connection = "AzureWebJobsStorage")] Stream myBlob, string name)
     {
-        _logger.LogInformation($"C# Blob trigger function Processed blob\n Name:{name} \n Size: {myBlob.Length} Bytes");
-
-        // Validate JSON
-        using (var reader = new StreamReader(myBlob))
+        try
         {
-            var jsonData = reader.ReadToEnd();
-            if (IsValidJson(jsonData))
+            _logger.LogInformation("C# HTTP trigger function ReceiveData received a request.");
+
+            var (episodeUrl, participantUrl) = GetConfigurationUrls();
+            if (string.IsNullOrEmpty(episodeUrl) || string.IsNullOrEmpty(participantUrl))
+            {
+                _logger.LogError("One or both URLs are not configured");
+                return;
+            }
+
+            string? fileName = name;
+
+            if (string.IsNullOrEmpty(fileName))
```

Comment:
Don't think we need to do this check any more since the name parameter from the blob trigger should always have a value

Reply:
removed this.